src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,22 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("User cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("User with this login already exists");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must contain 6 or more symbols");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserException("Password must contain 6 or more symbols");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new InvalidUserException("User must be at least 18 years old");
```

Comment:
This condition should be split into two separate checks: one for `user.getAge() == null` and another for `user.getAge() < 18`. This is to ensure that the error messages are informative and specific to the validation that failed.